posthog-core/src/index.ts

```diff
@@ -839,10 +840,11 @@ export abstract class PostHogCoreStateless {
 
     const payload = JSON.stringify(data)
 
+    const canGzip = this.captureMode === 'json' && isGzipSupported();
     const url =
       this.captureMode === 'form'
         ? `${this.host}/e/?ip=1&_=${currentTimestamp()}&v=${this.getLibraryVersion()}`
-        : `${this.host}/batch/`
+        : `${this.host}/batch/${canGzip ? '?compression=gzip-js' : ''}`
```

Comment:
not sure if we need `compression=gzip-js` here, I will need to check other SDKs.
we use `"Content-Encoding", "gzip"` header on Android/iOS but its not Web.

Reply:
Because we send it using compressed in the body we need to use it if I remember correctly.
Because content-encoding "gzip"  uses the native HTTP transport compression :smile:

Reply:
I think I found a few examples with `/s/?compression=gzip-js` URL but not the `/batch` so I don't know if the batch needs/accepts that.

Reply:
I migrated to Content-Encoding because it's a bit more cleaner approach and perhaps easier for some self-hosters as the reverse proxy likely handles the encoding before it hits PostHog.

`/batch/` does accept it, at least the OpenAPI documentation says so [here](https://github.com/PostHog/posthog/blob/df24f8d1156987ebd45e05877e77968845e8fe5c/openapi/capture.yaml#L144-L153)